pkg/entrypoint/entrypointer.go

```diff
@@ -264,6 +273,20 @@ func (e Entrypointer) readResultsFromDisk(ctx context.Context, resultDir string)
 			ResultType: result.TaskRunResultType,
 		})
 	}
+	for resultName, resultPath := range e.StepResults {
+		fileContents, err := os.ReadFile(resultPath)
+		if os.IsNotExist(err) {
+			continue
+		} else if err != nil {
+			return err
+		}
+		// if the file doesn't exist, ignore it
+		output = append(output, result.RunResult{
+			Key:        resultName,
+			Value:      string(fileContents),
+			ResultType: result.TaskRunResultType,
+		})
+	}
```

Comment:
Is this going to conflict with other Task level results? Should we be propagating these up like this? Should we be using a different ResultType instead?

Reply:
Following up from a Slack conversation: This was originally done under the expectation that we were going to bubble up step level results to the Task level. However

1. I don't think we should do this - I think we should make Task result exporting explicit so we're not overwhelming Task results with step-specific results without their context. e.g. if you have a git action that produces `url` and an oci action that produces `repo-url`, we shouldn't automatically bubble these up since it's going to be very difficult to understand the meaning of the results without additional details even though there's no naming collision (similar issues arise with other similar names - `sha`/`digest`, `url`/`uri`, `outcome`/`result`/`value`). @chitrangpatel is planning to follow up on this at the next WG.
2. Even if we do want to bubble these values up to the Task level, this decision should probably be made by the controller, not the entrypoint. There's still value in separating these out for the entrypoint so we can have a clear API to know what values were set where.

Reply:
For now, I will add a new result Type and let the controller logic filter things out.  I will limit this PR to only add results to `StepResults` that are explicitly requested via the `value` field in TaskResults. This will prevent auto-surfacing. 


After the discussion at the upcoming data interfaces WG, we will revisit auto-surfacing (if needed).

Reply:
I implemented this as suggested. PTAL.